src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.asm

```diff
@@ -317,4 +319,74 @@ ExchangeRetry
 
     LEAF_END RhpCheckedXchg
 
+;; RhpAssignRefArm64(Object** dst, Object* src)
+;;
+;; On entry:
+;;   x14  : the destination address (LHS of the assignment).
+;;   x15  : the object reference (RHS of the assignment).
+;;
+;; On exit:
+;;   x9   : trashed
+;;   x10  : trashed if WRITE_BARRIER_CHECK
+;;   x14  : trashed
+;;
+    LEAF_ENTRY RhpAssignRefArm64, _TEXT
+    ALTERNATE_ENTRY RhpAssignRefAVLocation
+    ALTERNATE_ENTRY RhpAssignRefX1AVLocation
+
+        stlr    x15, [x14]
+
+        INSERT_UNCHECKED_WRITE_BARRIER_CORE x14, x15
+
+        ret
+    LEAF_END RhpAssignRefArm64
+
+;; void JIT_CheckedWriteBarrier(Object** dst, Object* src)
+;; On entry:
+;;   x14 : the destination address (LHS of the assignment)
+;;   x15 : the object reference (RHS of the assignment)
+;;
+;; On exit:
+;;   x9  : trashed
+;;   x10 : trashed if WRITE_BARRIER_CHECK
+;;   x14 : incremented by 8 to implement JIT_ByRefWriteBarrier contract
+;;
+    LEAF_ENTRY RhpCheckedAssignRefArm64, _TEXT
+    ALTERNATE_ENTRY RhpCheckedAssignRefAVLocation
+    ALTERNATE_ENTRY RhpCheckedAssignRefX1AVLocation
+
+        stlr    x15, [x14]
+
+        ;; TODO: This trashes x14
```

Comment:
Is this TODO relevant? Do want to file issue on it to make sure we do not forget about it?

Reply:
I am going to fix the two `TODO` issues regarding trashing `x14` in the helpers I am adding. The Linux version avoids this issue by passing the register to trash to `INSERT_CHECKED_WRITE_BARRIER_CORE` macro.

I also placed two `TODO` comments regarding trashing `x10` if `WRITE_BARRIER_CHECK` is enabled in the existing helpers, `RhpCheckedLockCmpXchg` and `RhpCheckedXchg`. On closer inspection, there is no bug because `x10` is 'trashed' with `x0`'s value. I'll add a comment on that.